src/test/java/org/gridsuite/securityanalysis/server/SecurityAnalysisControllerTest.java

```diff
@@ -648,8 +648,8 @@ public void stopTest() throws Exception {
         // wait for security analysis to actually run before trying to stop it
         countDownLatch.await();
 
-        mockMvc.perform(put("/" + VERSION + "/results/" + RESULT_UUID + "/stop"
-            + "?receiver=me"))
+        mockMvc.perform(put("/" + VERSION + "/results/" + RESULT_UUID + "/stop" + "?receiver=me")
+                        .header(HEADER_USER_ID, "testUserId"))
```

Comment:
it seems here we manage to get the stopped notification and not the cancelfailed notification. Why ?

Reply:
i think because we running in different thread and we wait

Reply:
// wait for security analysis to actually run before trying to stop it
countDownLatch.await();

And because in this service we have a `SecurityAnalysisProviderMock` which make: 
```
// creating a long completable future which is here to be canceled
return new CompletableFuture<SecurityAnalysisReport>().completeOnTimeout(REPORT, 3, TimeUnit.SECONDS);
```
then we have the time to cancel the computation and obtain a clean sa.stopped

Both of those code parts seems absent of other computation services (LF, SE,Dy, CC etc...)
Should add it